code/modules/xenoarchaeology/xenoartifact_traits.dm

```diff
@@ -0,0 +1,1011 @@
+		return TRUE
+	..()
+
+/datum/xenoartifact_trait/activator/signal/calculate_charge(datum/source, obj/item/thing, mob/user, atom/target, params)
+	var/obj/item/xenoartifact/X = source
+	X.default_activate(charge, user, target)
+	log_game("[user]:[isliving(user) ? user?.ckey : "no ckey"] signalled [X] at [world.time]. [X] located at [X.x] [X.y] [X.z].")
+
+/datum/xenoartifact_trait/activator/batteryneed
+	desc = "Charged"
+	label_desc = "Charged: The material has a natural power draw. Supplying any current to this will cause a reaction."
+	charge = 25
+	signals = list(COMSIG_PARENT_ATTACKBY)
+
+/datum/xenoartifact_trait/activator/batteryneed/on_item(obj/item/xenoartifact/X, atom/user, atom/item)
+	if(istype(item, /obj/item/multitool))
+		to_chat(user, "<span class='info'>The [item.name] displays a draw of [X.charge_req].</span>")
+		return TRUE
+	..()
+
+/datum/xenoartifact_trait/activator/batteryneed/calculate_charge(datum/source, obj/item/thing, mob/user, atom/target, params)
+	var/obj/item/xenoartifact/X = source
+	if(istype(thing, /obj/item/stock_parts/cell))
+		var/obj/item/stock_parts/cell/C = thing
+		if(X.default_activate(charge, user, user))
+			C.charge -= X.charge_req*10
+
+//Minor traits - Use the to define aspects of the artifact without any immediate interaction
+/datum/xenoartifact_trait/minor/looped
+	desc = "Looped"
+	label_desc = "Looped: The Artifact feeds into itself and amplifies its own charge."
+
+/datum/xenoartifact_trait/minor/looped/on_item(obj/item/xenoartifact/X, atom/user, atom/item)
+	if(istype(item, /obj/item/multitool))
+		to_chat(user, "<span class='info'>The [item.name] displays a resistance reading of [X.charge_req*0.1].</span>") 
+		return TRUE
+	..()
+
+/datum/xenoartifact_trait/minor/looped/activate(obj/item/xenoartifact/X)
+	X.charge = ((100-X.charge)*0.2)+X.charge //This should generally cut off around 100
+
+/datum/xenoartifact_trait/minor/capacitive
+	desc = "Capacitive"
+	label_desc = "Capacitive: The Artifact's structure allows it to hold extra charges."
+	var/charges //Extra uses, not total
+	var/saved_cooldown //This may be considered messy but it's a more practical approach that avoids making an edgecase
+
+/datum/xenoartifact_trait/minor/capacitive/on_init(obj/item/xenoartifact/X)
+	charges = pick(0, 1, 2) //Extra charges, not total
+
+/datum/xenoartifact_trait/minor/capacitive/on_touch(obj/item/xenoartifact/X, mob/user)
+	to_chat(user, "<span class='notice'>The hairs on your neck stand up after touching the [X.name].</span>")
+	return TRUE
+
+/datum/xenoartifact_trait/minor/capacitive/activate(obj/item/xenoartifact/X)
+	if(!(saved_cooldown) && X.cooldown)
+		saved_cooldown = X.cooldown //Avoid doing this on init beacause malfunctions can change it in the future
+	if(charges)
+		charges -= 1
+		X.cooldown = -1000 SECONDS //This is better than making a unique interaction in xenoartifact.dm
```

Comment:
Could you explain this?

Reply:
If the artifact has an extra use from capacitive, the base cooldown will be -1000, so traits that add additional cooldown, like time-stop, will not start the timer. Then when all the extra charges are used, max two extra, the timer will go back to normal and run the intended time. After that a random amount of extra charges will generate, (1, 2)